### PR TITLE
chore(cmf): add setState in replacement of updateState

### PR DESCRIPTION
**Pull request by jmfrancois**
updateState still exist with a warning so no break

**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

to udpate the state of a container you have to use this.props.updateState.
With react it's this.setState

**What is the new behavior?**

lets switch to this.props.setState
a warning is done in the console if you are using updateState;

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://359.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://359.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://359.talend.surge.sh)

File: packages/cmf/src/componentState.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
 import { PropTypes } from 'react';
 import { Map } from 'immutable';
-import { actions } from 'react-cmf';
+import actions from './actions';
 
 export function getStateAccessors(dispatch, name, id, DEFAULT_STATE = new Map()) {
 	return {
 		updateState(state) {
+			console.warn('DEPRECATION WARNING: please use props.setState');
+			dispatch(
```
Comment (jsomsanith) [inline packages/cmf/src/componentState.js]: Can you call setState instead of duplicating its code ?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://359.talend.surge.sh)

File: packages/cmf/src/componentState.js (hunk 1 above)
Comment (jmfrancois) [inline packages/cmf/src/componentState.js]: done

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://359.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://359.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://359.talend.surge.sh)

**Pull request merged**